packages/dataviews/src/types.ts

```diff
@@ -40,40 +40,103 @@ export interface Field {
 	/**
 	 * The width of the field column.
 	 */
-	width?: string | number;
+	width: string | number | undefined;
 
 	/**
 	 * The minimum width of the field column.
 	 */
-	maxWidth?: string | number;
+	maxWidth: string | number | undefined;
 
 	/**
 	 * The maximum width of the field column.
 	 */
-	minWidth?: string | number;
+	minWidth: string | number | undefined;
 
 	/**
 	 * Whether the field is sortable.
 	 */
-	enableSorting?: boolean;
+	enableSorting: boolean | undefined;
 
 	/**
 	 * Whether the field is searchable.
 	 */
-	enableGlobalSearch?: boolean;
+	enableGlobalSearch: boolean | undefined;
 
 	/**
 	 * Whether the field is filterable.
 	 */
-	enableHiding?: boolean;
+	enableHiding: boolean | undefined;
 
 	/**
 	 * The list of options to pick from when using the field as a filter.
 	 */
-	elements?: Option[];
+	elements: Option[] | undefined;
 
 	/**
 	 * Filter config for the field.
 	 */
-	filterBy?: filterByConfig;
+	filterBy: filterByConfig | undefined;
+}
+
+export type NormalizedField = Required< Field >;
+
+export type Data = Item[];
+
+export interface Filter {
+	/**
+	 * The field to filter by.
+	 */
+	field: string;
+
+	/**
+	 * The operator to use.
+	 */
+	operator: string;
+
+	/**
+	 * The value to filter by.
+	 */
+	value: any;
+}
+
+export interface View {
+	/**
+	 * The layout of the view.
+	 */
+	type: string;
```

Comment:
It looks like this could also be a union of constants:

```suggestion
	type: 'table' | 'grid' | 'list';
```

Reply:
I think this list is going to be dynamic (registering layouts) ultimately.